### PR TITLE
Improve documentation 1 to implement 1

### LINKED ISSUE #32: The anchors are not defined well
**Describe the feature or change you'd like**
Currently the anchors defined in client do not meet the requirements of snek-tonic. Therefore, the package imports should be revised throughout the code.

**Additional context**

### LINKED ISSUE #33: The code should be adapted to the new coding guideline
**Describe the feature or change you'd like**
Due to the change in the coding guideline on Friday, 8. May 2020 the code should be adjusted over time. 

**Additional context**
Ref: snek-at/tonic/pull/1

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the anchor definitions do not follow our standard. #32 
- Currently the documentation does not follow our standard. #33

**What is the new behavior (if this is a feature change)?**
- Now the anchor definitions do follow our standard. Closes #32 
- Now the documentation does follow our standard. Closes #33

**Other information**:
- Ref: https://githb.com/snek-at/tonic/
- Ref: 🐍

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -3,6 +3,12 @@
     "editor.fontLigatures": true,
     "editor.snippetSuggestions": "top",
     "editor.rulers": [80, 120],
+    // Set the default
```
Comment (Aichnerc) [inline .vscode/settings.json]:
```suggestion
     "_comment": "Set the default",
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -3,6 +3,12 @@
...
+    "editor.formatOnSave": false,
+    // Enable per-language
```
Comment (Aichnerc) [inline .vscode/settings.json]:
```suggestion
    "_comment": "Enable per-language",
```

File: src/templates/snek/gql/tasks/general.ts [hunk 3]
```diff
@@ -1,45 +1,53 @@
...
-import { ISnekGqlTemplate } from "../index";
+import { SnekGqlTemplate } from "../index";
 //#endregion
 
 //#region > Interfaces
-/** @interface GitlabServerResponse defines the structure of the gitlab server response. */
-interface IGitlabServerResponse extends IResponse {
+/**
+ * @interface GitlabServerResponse defines the overall structure of a Gitlab
+ *                                server response from the SNEK-engine.
```
Comment (Aichnerc) [inline src/templates/snek/gql/tasks/general.ts]: Is this indentation on point?

File: src/templates/snek/gql/tasks/general.ts [hunk 4]
```diff
@@ -1,45 +1,53 @@
...
+ */
+interface GitlabServerResponse extends Response {
   data: { page: GitlabServerData };
 }
 
-/** @interface GitlabServerData defines the structure of the gitlab server data. */
+/**
+ * @interface GitlabServerData defines the structure of the specific data a
+ *                      GitlabServerResponse contains.
```
Comment (Aichnerc) [inline src/templates/snek/gql/tasks/general.ts]: Is this indentation on point?

File: src/templates/snek/gql/tasks/index.ts [hunk 5]
```diff
@@ -12,8 +12,11 @@ import SnekGqlUserTasks from "./user";
 //#endregion
 
 //#region > Interfaces
-/** @interface Response defines the basic response structure. */
-interface IResponse {
+/**
+ * @interface Response defines the basic response structure which fits all
+ *                       graphql requests.
```
Comment (Aichnerc) [inline src/templates/snek/gql/tasks/index.ts]: Is this indentation on point?

**Review by Aichnerc — CHANGES_REQUESTED** [review]
🥐 As crooked as this 𝖈𝖗𝖔𝖎𝖘𝖘𝖆𝖓𝖙.

File: .vscode/settings.json (hunk 1 above)
Comment (schettn) [reply .vscode/settings.json]: This is not needed here. But it is a good idea for default json files. Thx.

File: src/endpoints/index.ts [hunk 6]
```diff
@@ -1,13 +1,14 @@
 //#region > Imports
 //> Interfaces
-//#INSTALL "graphql"
+//#PACKAGE "graphql"
+//## nmp install "graphql"@14.6.0
```
Comment (pinterid) [inline src/endpoints/index.ts]: Change "nmp" to "npm"

File: src/endpoints/scraper.ts [hunk 7]
```diff
@@ -57,9 +60,11 @@ class Scraper implements ScraperEndpoint {
   }
 
   /**
-   * @param path Path to the endpoint. Specify it like "/foo/bar". The correct placement of the slashes is essential!
-   * @returns {object} DOM object.
-   * @description Get DOM object from specified path.
+   * Get DOM object from specified path
```
Comment (pinterid) [inline src/endpoints/scraper.ts]: Missing dot at the end of line.

File: src/session/index.ts [hunk 8]
```diff
@@ -1,12 +1,12 @@
 //#region > Imports
-//> JS Cookie
-//#INSTALL "js-cookie"
+//#PACKAGE "js-cookie"
+//## nmp install "graphql"@2.2.1
```
Comment (pinterid) [inline src/session/index.ts]: Change "nmp" to "npm"

File: src/session/sessions.ts [hunk 9]
```diff
@@ -1,10 +1,10 @@
 //#region > Imports
-//> Interfaces
-//#INSTALL "graphql"
+//#PACKAGE "graphql"
+//## nmp install "graphql"@14.6.0
```
Comment (pinterid) [inline src/session/sessions.ts]: Change "nmp" to "npm"

File: src/session/sessions.ts [hunk 10]
```diff
@@ -1,10 +1,10 @@
...
 // Contains the interface for gql queries, mutations and subscriptions
 import { DocumentNode } from "graphql";
-//> JS Cookie
-//#INSTALL "js-cookie"
+//#PACKAGE "js-cookie"
+//## nmp install "graphql"@2.2.1
```
Comment (pinterid) [inline src/session/sessions.ts]: Change "nmp" to "npm"

File: src/session/sessions.ts [hunk 11]
```diff
@@ -157,9 +157,8 @@ class SnekSession extends Session {
   }
 
   /**
-   * Send query:
+   * Send a query to the endpoint.
    *
-   * @description Send a query to the endpoint.
    * @param {string} token A authentication token.
```
Comment (pinterid) [inline src/session/sessions.ts]: Remove dot.

File: src/session/sessions.ts [hunk 12]
```diff
@@ -157,9 +157,8 @@ class SnekSession extends Session {
...
    * @param {DocumentNode} data A DocumentNode with a query.
```
Comment (pinterid) [inline src/session/sessions.ts]: Remove dot.

File: src/session/sessions.ts [hunk 13]
```diff
@@ -157,9 +157,8 @@ class SnekSession extends Session {
...
    * @param {object} variables A object with variables.
```
Comment (pinterid) [inline src/session/sessions.ts]: Remove dot,

File: src/templates/snek/gql/mutations/jwtAuth.ts [hunk 14]
```diff
@@ -1,18 +1,18 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## nmp install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/mutations/jwtAuth.ts]: Change "nmp" to "npm"

File: src/templates/snek/gql/mutations/user.ts [hunk 15]
```diff
@@ -1,18 +1,19 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## nmp install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/mutations/user.ts]: Change "nmp" to "npm"

File: src/templates/snek/gql/queries/general.ts [hunk 16]
```diff
@@ -1,17 +1,17 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## nmp install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/queries/general.ts]: Change "nmp" to "npm"

File: src/templates/snek/gql/queries/user.ts [hunk 17]
```diff
@@ -1,18 +1,18 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## nmp install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/queries/user.ts]: Change "nmp" to "npm" and remove one apostrophe.

File: src/templates/snek/gql/queries/user.ts [hunk 18]
```diff
@@ -29,11 +29,11 @@ const whoami = gql`
 
 /**
  * Get user profile:
```
Comment (pinterid) [inline src/templates/snek/gql/queries/user.ts]: Change ":" to "."

File: src/templates/snek/gql/tasks/user.ts [hunk 19]
```diff
@@ -1,47 +1,65 @@
...
+// Contains the SNEK template
+import { SnekGqlTemplate } from "../index";
 //#endregion
 
 //#region > Interfaces
-/** @interface RegistrationResponse defines the structure of a registration response. */
-interface IRegistrationResponse extends IResponse {
+/**
+ *  @interface RegistrationResponse defines the overall structure of a
+ *                                  registration response from the SNEK-engine
```
Comment (pinterid) [inline src/templates/snek/gql/tasks/user.ts]: Missing dot at the end of line.

File: src/templates/snek/gql/tasks/user.ts [hunk 20]
```diff
@@ -105,13 +128,13 @@ class SnekGqlUserTasks extends TaskError {
   }
 
   /**
-   * Cache a user
-   *
-   * @returns {Promise<ICacheResponse>} A JWT token.
+   * Cache a user.
```
Comment (pinterid) [inline src/templates/snek/gql/tasks/user.ts:131]: Missing newline.

**Review by pinterid — CHANGES_REQUESTED** [review]
Looks good after some changes.

**Comment by pinterid** [comment]
The type of change has to be set before merging!

File: src/templates/snek/gql/mutations/jwtAuth.ts [hunk 21]
```diff
@@ -1,18 +1,18 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## npm install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/mutations/jwtAuth.ts]: There is on " to much.

File: src/templates/snek/gql/mutations/user.ts [hunk 22]
```diff
@@ -1,18 +1,19 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## npm install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/mutations/user.ts]: There is one " to much.

File: src/templates/snek/gql/queries/general.ts [hunk 23]
```diff
@@ -1,17 +1,17 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## npm install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/queries/general.ts]: There is on " to much.

File: src/templates/snek/gql/queries/user.ts [hunk 24]
```diff
@@ -1,18 +1,18 @@
 //#region > Imports
-//> Graphql
-//#INSTALL "graphql-tag"
+//#PACKAGE "graphql-tag"
+//## npm install ""graphql-tag"@2.10.3
```
Comment (pinterid) [inline src/templates/snek/gql/queries/user.ts]: There is one " to much.

**Review by pinterid — CHANGES_REQUESTED** [review]
Looks good after some changes.

**Review by pinterid — APPROVED** [review]
Looks good to merge.

**Review by Aichnerc — APPROVED** [review]
LGTM 🥞

File: src/endpoints/apollo.ts [hunk 25]
```diff
@@ -79,6 +77,18 @@ class Apollo implements ApolloEndpoint {
   }
 
   //> Methods
+  /**
+   * Send: Provides requests for various graphql types.
+   *
+   * @param {string} type The type of the action you want to perform.
```
Comment (kleberbaum) [inline src/endpoints/apollo.ts]: Please remove the period.

File: src/endpoints/index.ts [hunk 26]
```diff
@@ -18,24 +19,27 @@ interface IEndpoint {
...
 interface ApolloEndpoint extends IEndpoint {
   /**
    * Send: Provides requests for various graphql types.
    *
-   * @param {string} type The type of the action you want to perform. Query, Mutation,...
-   * @param {DocumentNode} data The query structure.
-   * @param {object} variables A object which contains variables for the query structure.
-   * @param {object} headers Optional headers which get appended to the endpoint headers.
-   * @returns {Promise<object>} Resolved apollo data object.
+   * @param {string} type The type of the action you want to perform.
```
Comment (kleberbaum) [inline src/endpoints/index.ts]: Please remove the period.

File: src/index.ts [hunk 27]
```diff
@@ -18,27 +18,37 @@ import { IMainTemplate } from "./templates/index";
 //#endregion
 
 //#region > Interfaces
-/** @interface Endpoint defines the structure of object a endpoint requieres to initialize. */
+/**
+ * @interface Endpoint defines the structure of object a endpoint requirers to
+ *                     initialize.
```
Comment (kleberbaum) [inline src/index.ts:23]: Please fix indentation.

File: src/index.ts [hunk 28]
```diff
@@ -18,27 +18,37 @@ import { IMainTemplate } from "./templates/index";
...
-/** @interface Client will define the client structure. */
+/** @interface IClient will define the client structure */
 interface IClient {}
 //#endregion
 
 //#region > Classes
-/** @class The snek-client. Enjoy it. Will be implemented in the future. */
+/**
+ * @class The SNEK-client. Enjoy it. Will be implemented in the future
+ * @todo Rework the url checker. Add documentation
```
Comment (kleberbaum) [inline src/index.ts]: and add documentation

File: src/session/sessions.ts [hunk 29]
```diff
@@ -63,20 +61,18 @@ class GithubSession extends Session {
   }
 }
 
-/** @class A Snek SubSession. */
+/** @class A SNEK SubSession */
 class SnekSession extends Session {
   refreshTokenName: string = "refresh";
-
```
Comment (kleberbaum) [inline src/session/sessions.ts:69]: Please add a newline.

File: src/templates/snek/gql/tasks/auth.ts [hunk 30]
```diff
@@ -10,17 +10,23 @@ import { TaskError } from "../errors";
...
+ *                         response from the SNEK-engine.
+ */
 interface IAuthResponse extends IResponse {
   data: { auth: AuthData };
 }
 
-/** @interface AuthData defines the structure of the specific data a authentication response contains. */
+/**
+ * @interface AuthData defines the structure of the specific data a
+ *                      authentication response contains.
```
Comment (kleberbaum) [inline src/templates/snek/gql/tasks/auth.ts]: Please fix indentation.

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge after a few changes.

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**